src/Common/FunctionDocumentation.cpp

```diff
@@ -22,12 +22,121 @@ VersionNumber VERSION_UNKNOWN = {0};
+            result += "`](/sql-reference/data-types/tuple)";
+        else if (type.starts_with("Map")) /// matches 'Map(K, V)'
+            result += "`](/sql-reference/data-types/map)";
+        else if (type.starts_with("Variant")) /// matches 'Variant(T1, T2, ...)', 'Variant(UInt8, String)' etc
+            result += "`](/sql-reference/data-types/variant)";
+        else if (type.starts_with("LowCardinality")) /// matches 'LowCardinality(T)', 'LowCardinality(UInt8)', 'LowCardinality(String)' etc
+            result += "`](/sql-reference/data-types/lowcardinality)";
+        else if (type.starts_with("Nullable")) /// matches 'Nullable(T)', 'Nullable(UInt8)', 'Nullable(String)' etc
+            result += "`](/sql-reference/data-types/nullable)";
+        else if (type.starts_with("AggregateFunction")) /// matches 'AggregateFunction(agg_func, T)', 'AggregateFunction(any, UInt8)' etc
+            result += "`](/sql-reference/data-types/aggregatefunction)";
+        else if (type.starts_with("SimpleAggregateFunction")) /// matches 'SimpleAggregateFunction(agg_func, T)', 'SimpleAggregateFunction(any, UInt8)' etc
+            result += "`](/sql-reference/data-types/simpleaggregatefunction)";
+        else if (type == "Point")
+            result += "`](/sql-reference/data-types/geo#point)";
+        else if (type == "Ring")
+            result += "`](/sql-reference/data-types/geo#ring)";
+        else if (type == "LineString")
+            result += "`](/sql-reference/data-types/geo#linestring)";
+        else if (type == "MultiLineString")
+            result += "`](/sql-reference/data-types/geo#multilinestring)";
+        else if (type == "Polygon")
+            result += "`](/sql-reference/data-types/geo#polygon)";
+        else if (type == "MultiPolygon")
+            result += "`](/sql-reference/data-types/geo#multipolygon)";
+        else if (type == "Expression")
+            result += "`](/sql-reference/data-types/special-data-types/expression)";
+        else if (type == "Set")
+            result += "`](/sql-reference/data-types/special-data-types/set)";
+        else if (type == "Nothing")
+            result += "`](/sql-reference/data-types/special-data-types/nothing)";
+        else if (type == "Interval")
+            result += "`](/sql-reference/data-types/special-data-types/interval)";
+        else if (type.starts_with("Nested")) /// matches 'Nested(N1 T1, N2 T2, ...)' etc
+            result += "`](/sql-reference/data-types/nested-data-structures/nested)";
+        else if (type == "Dynamic")
+            result += "`](/sql-reference/data-types/dynamic)";
+        else if (type == "JSON")
+            result += "`](/sql-reference/data-types/newjson)";
+        else if (type == "Lambda")
+            result += "`](/sql-reference/functions/overview#arrow-operator-and-lambda)";
+        else
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected data type: {}", type);
```

Comment:
Above mapping looks like it covers all cases. If a case is missing, then this exception will tell us whenever someone SELECTs from `system.functions`.